bin/install-macosx_arm64

```diff
@@ -1,16 +1,29 @@
 #!/bin/bash
 
-# Install scripts for M1 Macs
-# See https://github.com/PostHog/posthog/issues/2916
-# NB: use cryptography==3.4.7
+## NOTE: This is a helper script to simplify the process of getting setup on macOS. If in doubt check https://posthog.com/handbook/engineering/developing-locally
 
-# Set ld flags to use OpenSSL installed with brew
-export LDFLAGS="-L$(brew --prefix openssl)/lib"
-export CPPFLAGS="-I$(brew --prefix openssl)/include"
+set -e
 
-# Use system OpenSSL instead of BoringSSL for GRPC
-export GRPC_PYTHON_BUILD_SYSTEM_OPENSSL=1
-export GRPC_PYTHON_BUILD_SYSTEM_ZLIB=1
+echo "Attempting to install all dependencies and setup virtualenv..."
+echo ""
 
-pip cache purge
-pip install -r requirements.txt
+# Ensure we are in a directory ending with "posthog"
+if [[ "$PWD" != *"/posthog" ]]; then
+    echo "Please run this script from a directory ending with 'posthog'"
+    exit 1
+fi
+
+if [[ $* == *--reset* ]]; then
+    rm -rf env/
+fi
+
+python3.10 -m venv env
+source env/bin/activate
```

Comment:
I think this is a bit too opinionated, e.g. I prefer `pyenv-virtualenv`, which activates the virtualenv automatically whenever my working directory is inside the `posthog` repo

Reply:
But it doesn't matter for the installation right? It doesn't actually activate the env for your terminal, just for this script

Reply:
Hmm, thinking about this, the development guide only mentions `pyenv` and not `pyenv-virtualenv`, so makes sense to be in line with the "canonical" approach here